### PR TITLE
Packaging: Add necessary files

### PR DESCRIPTION
**Pull request by velurimithun**
- Manifest for flatpak
**Reference:** http://docs.flatpak.org/en/latest/index.html

### CODE REVIEW

File: flatpak/org.xorg.adriconf.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+			]
+		}
+	],
+
+	"finish-args": [
+		/*In future we need wayland also!*/
+   		"--socket=x11",
+   		"--share=ipc",
+		"--device=dri",
+		"--talk-name=org.freedesktop.secrets",
```
Comment (jlHertel) [inline flatpak/org.xorg.adriconf.json]: What exactly are this used for?
If I got it right, this option is only used when you need to talk to other processed using IPC, which I think is not needed in this case.

Comment (velurimithun) [reply flatpak/org.xorg.adriconf.json]: We are getting org.freedesktop.Dbus error

**Comment by velurimithun** [comment]
We can also add cleanup: paths to the submodules but I'm unclear about what and all files can be cleared.

I read that it reduces the package size, we need to add it

**Comment by velurimithun** [comment]
We also need rw: permission since we use user-wide, system-wide and source code in mesa driver files.

But, when I use **filesystem=host** option for this I was getting an error

File: flatpak/org.xorg.adriconf.json (hunk 1 above)
Comment (velurimithun) [reply flatpak/org.xorg.adriconf.json]: Failed to register: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown

Comment (velurimithun) [reply flatpak/org.xorg.adriconf.json]: Although I use those last 2 commands I'm getting error

**Comment by AdrianKoshka** [comment]
Changing the build-system to `cmake-ninja` (Don't know off hand whether you use ninja or not) seems to yield the same result:

```
[21/21] Linking CXX executable adriconf
[0/1] Install the project...
-- Install configuration: ""
-- Installing: /app/share/locale/en/LC_MESSAGES/adriconf.mo
-- Installing: /app/share/locale/hr/LC_MESSAGES/adriconf.mo
-- Installing: /app/share/locale/pt_BR/LC_MESSAGES/adriconf.mo
Committing stage build-adriconf to cache
Cleaning up
Committing stage cleanup to cache
Finishing app
Error: Command 'adriconf' not found
```

**Comment by AdrianKoshka** [comment]
Hypothetically, one could do:

```json
{
            "name": "adriconf",
            "buildsystem": "simple",
	    "build-commands": [
		    "mkdir build-dir",
		    "cd build-dir",
		    "cmake ..",
		    "make",
		    "make install"
	    ],
            "sources": [
                {
                    "type": "git",
                    "url": "https://github.com/jlHertel/adriconf",
                    "branch": "master"
                }
            ]
        }
```

But it does feel like a hack, seeing as flatpak-builder can orchestrate projects that use `cmake` normally...

**Comment by AdrianKoshka** [comment]
```
========================================================================
Building module adriconf in /home/ttubbs/flathub/.flatpak-builder/build/adriconf-5
========================================================================
Already on 'master'
Running: mkdir build-dir && cd build-dir && cmake .. && make && make install
-- The C compiler identification is GNU 8.2.0
-- The CXX compiler identification is GNU 8.2.0
-- Check for working C compiler: /run/ccache/bin/cc
-- Check for working C compiler: /run/ccache/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /run/ccache/bin/c++
-- Check for working CXX compiler: /run/ccache/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.5.2") 
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
-- Checking for module 'gtkmm-3.0'
--   Found gtkmm-3.0, version 3.22.2
CMake Warning at /usr/share/cmake-3.12/Modules/FindBoost.cmake:577 (message):
  Imported targets and dependency information not available for Boost version
  (all versions older than 1.33)
Call Stack (most recent call first):
  /usr/share/cmake-3.12/Modules/FindBoost.cmake:959 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.12/Modules/FindBoost.cmake:1618 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:37 (find_package)


CMake Error at /usr/share/cmake-3.12/Modules/FindBoost.cmake:2044 (message):
  Unable to find the requested Boost libraries.

  Unable to find the Boost header files.  Please set BOOST_ROOT to the root
  directory containing Boost or BOOST_INCLUDEDIR to the directory containing
  Boost's headers.
Call Stack (most recent call first):
  CMakeLists.txt:37 (find_package)


-- Checking for module 'libxml++-2.6'
--   Found libxml++-2.6, version 2.40.1
-- Checking for module 'x11'
--   Found x11, version 1.6.5
-- Checking for module 'libdrm'
--   Found libdrm, version 2.4.92
-- Checking for module 'libpci'
--   Found libpci, version 3.6.2
-- Found Intl: /usr/include  
-- Found Gettext: /usr/bin/msgmerge (found version "0.19.8.1") 
CMake Error at /usr/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.12/Modules/FindGTest.cmake:196 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:95 (find_package)


CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
Boost_INCLUDE_DIR (ADVANCED)
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf
   used as include directory in directory /run/build/adriconf

-- Configuring incomplete, errors occurred!
See also "/run/build/adriconf/build-dir/CMakeFiles/CMakeOutput.log".
Error: module adriconf: Child process exited with code 1
```
This is the error I get with your build instructions...so yeah, my hack doesn't work.

File: flatpak/org.x.adriconf.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			adriconf (Advanced DRI CONFigurator) is a GUI tool used to configure open source graphics drivers. It works by setting options and writing them to the standard drirc file used by the Mesa drivers.
+		</p>
+		<p>
+			GUI Tool used to configure OpenGL drivers. You can use it to optimize game settings or even workaround issues with them
+		</p>
+	</description>
+	<launchable type="desktop-id">org.x.adriconf.desktop</launchable>
+	<screenshots>
+		<screenshot type="default">
+			<image type="source">https://github.com/jlHertel/adriconf/blob/master/drilogo.jpg?raw=true</image>
```
Comment (velurimithun) [inline flatpak/org.x.adriconf.appdata.xml]: After this is merged, I'll replace these with 2 screenshots url from this repo

**Comment by jlHertel** [comment]
@velurimithun In general the code seems pretty good to me.
My only complaint is about the commit which fixes the indentation. Can you kindly squash it together with the commit before it? So that the commit history stays a little bit more clean.
- Leave me a note in case you need any clarification with how the squash and rebase works.

I'm not very sure how flatpak works, but if I got it right we are going to just submit the configuration file to the flathub and they will download and build the package?
In case we need to provide the complete package already built, would it make sense to also add this to the jenkins build? So we don't need to keep generating those by hand.

**Comment by velurimithun** [comment]
We need to make a release with the updated files and include that version **tag** in the manifest.
Since it is a requirement to use a released version for the flatpak build.

Once you release a version I'll update that [here in flathub PR](https://github.com/flathub/flathub/pull/605)

**Comment by jlHertel** [comment]
OK, so this only leaves us with the commit that needs to be squashed.
Can you kindly squash it?

**Comment by velurimithun** [comment]
I've suquashed it already!

Indentation one is not there pl. Check

Do you want me to squash any other commit?

**Pull request merged**